File: src/renderer/src/types/portforward.ts
export interface TunnelConfig {
  local_port: number
  remote_host: string
  remote_port: number
  binding_address: string
}

export interface RemoteTunnelConfig {
  remote_port: number
  local_host: string
  local_port: number
  binding_address: string
}

export interface DynamicTunnelConfig {
  local_port: number
  binding_address: string
}

export interface DynamicTunnelConfig {
  local_port: number
  binding_address: string
}

export interface TunnelInfo {
  id: string
  connection_id: string
  name: string
  type: 'local' | 'remote' | 'dynamic'
  local_port: number
  remote_host: string
  remote_port: number
  status: 'active' | 'stopped' | 'error'
  error?: string
}

export interface CreateTunnelRequest {
  type: 'local' | 'remote' | 'dynamic'
  connection_id: string
  name: string
  config?: TunnelConfig
  remote?: RemoteTunnelConfig
  dynamic?: DynamicTunnelConfig
}

export interface PortForwardConfig {
  id: string
  name: string
  connection_id: string
  type: 'local' | 'remote' | 'dynamic'
  local_port: number
  remote_host: string
  remote_port: number
  binding_address: string
  auto_start: boolean
}

File: backend/internal/models/portforward.go
package models

type TunnelConfig struct {
	LocalPort      int    `json:"local_port"`
	RemoteHost     string `json:"remote_host"`
	RemotePort     int    `json:"remote_port"`
	BindingAddress string `json:"binding_address"`
}

type RemoteTunnelConfig struct {
	RemotePort     int    `json:"remote_port"`
	LocalHost      string `json:"local_host"`
	LocalPort      int    `json:"local_port"`
	BindingAddress string `json:"binding_address"`
}

type DynamicTunnelConfig struct {
	LocalPort      int    `json:"local_port"`
	BindingAddress string `json:"binding_address"`
}

type TunnelInfo struct {
	ID           string `json:"id"`
	ConnectionID string `json:"connection_id"`
	Name         string `json:"name"`
	Type         string `json:"type"` // "local" or "remote"
	LocalPort    int    `json:"local_port"`
	RemoteHost   string `json:"remote_host"`
	RemotePort   int    `json:"remote_port"`
	Status       string `json:"status"` // active, stopped, error
	Error        string `json:"error,omitempty"`
}

type CreateTunnelRequest struct {
	Type         string              `json:"type"` // "local", "remote", or "dynamic"
	ConnectionID string              `json:"connection_id"`
	Name         string              `json:"name"`
	Config       TunnelConfig        `json:"config,omitempty"`
	Remote       RemoteTunnelConfig  `json:"remote,omitempty"`
	Dynamic      DynamicTunnelConfig `json:"dynamic,omitempty"`
}

type StopTunnelRequest struct {
	ConnectionID string `json:"connection_id"`
	TunnelID     string `json:"tunnel_id"`
}
